src/store/queue/sagas.ts

```diff
@@ -167,21 +174,22 @@ export function* watchPlay() {
         track: playActionTrack
       })
 
-      const user: User = playActionTrack
+      const user: User | null = playActionTrack
         ? yield select(getUser, { id: playActionTrack.owner_id })
         : null
 
       // Skip deleted tracks
       if (
         (playActionTrack && playActionTrack.is_delete) ||
+        // @ts-ignore this is technically impossible with typing, but keeping for now to avoid potential runtime issues
```

Comment:
better to do a ! operator w/ a comment?

Reply:
Maybe I'm missing it but why is it impossible? Can't `user` be null?

Reply:
its a similar case to the ts demo today: user is type `never` here since its set to null if playActionTrack is false, which if looking at the if statement, will always be null since (playActionTrack && playActionTrack.is_delete) would only pass execution via || if it was false, therefore saying `user` would be type null, so checking `user.is_deactivated` is not a possible case.

ultimately, we can probably jut remove this || check, but wanted to keep the code just to make sure i dont break runtime execution, since the types are in a funky enough state that they may not be reliable till we get the `typed-redux-saga` stuff going. Thoughts?

Reply:
Can't the `select()` be undefined/null though if hte user isn't in the store?

Reply:
so it boils down to `playActionTrack`, when that i falsy, user is null, so `(playActionTrack || user.is_deactivated)`, -> `(false || null.is_deactivated)` so ts types become `(boolean || never)`

Reply:
I don't think I follow. `playActionTrack` can be true, `getUser` could return undefined (so `user` is undefined) and `playActionTrack.is_delete` can be `false` all at the same time, no?

`if ((true && false) || undefined)`

Reply:
oh youre so right! will update the comment for now. looks like user type is just "null", which is happening for unknown reasons atm, but can address when we convert to typed-redux-saga